src/ScottPlot5/ScottPlot5/DataSources/OHLCSourceXsYs.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ScottPlot.DataSources;
+
+public class OHLCSourceXsYs : IOHLCSource
+{
+    private readonly IReadOnlyList<double> Xs;
+    private readonly IReadOnlyList<OHLC> Ys;
+
+    public OHLCSourceXsYs(IReadOnlyList<double> xs, IReadOnlyList<OHLC> ys)
+    {
+        Xs = xs;
+        Ys = ys;
+    }
+
+    private Tuple<double, OHLC> GetCoordinatesAt(int index)
+    {
+        return new (Xs[index], Ys[index]);
+    }
+
+    public IReadOnlyList<Tuple<double, OHLC>> GetOHLCPoints()
+    {
```

Comment:
The use of Zip is more obvious
```c#
  return Xs.Zip(Ys, (x, y) => (x, y)).ToArray();
```

Reply:
I changed it, but I believe this is copy-pasted from `ScatterSourceXsYs`. I changed it there too but it may have made its way elsewhere.

Reply:
I always forget about Zip! Thanks for the reminder 😄 

https://learn.microsoft.com/en-us/dotnet/api/system.linq.enumerable.zip